agent/agent_test.go

```diff
@@ -155,6 +156,23 @@ func TestRunProducts(t *testing.T) {
 	assert.NotNil(t, a.products["host"], "product is under \"host\" key")
 }
 
+func TestAgent_RecordManifest(t *testing.T) {
+	t.Run("adds to MetadataSeekers when seekers exist", func(t *testing.T) {
```

Comment:
Not sure if it's worthwhile for now or something for the backlog, but would this be a candidate for a table test where we could verify any failure cases as well? (We'd probably have to do some mocking in that case too.)

Reply:
That's a good q - I ran through an implementation of this before pushing the branch up and found that the agent setup code was too elaborate. It worked, but implementing more cases with the pattern highlighted the agent/product/seeker boundary issues more than they tested semantics. We'll fix this problem in a better way down the line.